frontend/trip-planner/src/Components/TripAddUpdate.js

```diff
@@ -0,0 +1,146 @@
+            if (result.status === 401) {
+                navigate('/login');
+            } else {
+                if (result.status === 200) {
+                    return result;
+                } else {
+                    setInsertFailed(true);
+                }
+            };
+        } catch (e) {
+            navigate('/error');
+        }
+    };
+
+
+    async function getATrip(id) {
+        try {
+            const tripDB = await getSpecificTrip(id);
+            if (tripDB.status === 401) {
+                navigate('/login');
+            } else {
+                const jsonData = await tripDB.json();
+                setTrip(jsonData);
+                setCountry(options.filter(obj => obj.label === jsonData.country)[0]);
+                setCity(jsonData.city);
+                const emailsDB = jsonData.userData.map(el => el.username);
+                setEmails(emailsDB);
+                setStartDate(new Date(jsonData.start_date));
+                setEndDate(new Date(jsonData.end_date));
+            };
+        } catch (e) {
+            navigate('/error');
+        }
+    };
+
+
+    useEffect(() => {
+        if (!props.isTripAdd) {
+            getATrip(props.trip.id);
+        }
+    }, []);
+
+
+    async function updateUserTrip(updatedTrip) {
+        try {
+            const result = await updateTrip(updatedTrip, props.trip.id);
+            if (result.status === 401) {
+                navigate('/login');
+            } else {
+                if (result.status === 200) {
+                    return result;
+                } else {
+                    setUpdateFailed(true);
+                }
+            };
+        } catch (e) {
+            navigate('/error');
+        }
+    };
+
+
+    async function submitTrip(e) {
+        e.preventDefault();
+        if (!country.label || !city || !startDate || !endDate || !emails) {
+            setFieldsFilled(true);
+        } else {
+            const newTrip = {
+                country: country.label,
+                city: city,
+                start_date: startDate,
+                end_date: endDate,
+                emails: emails
+            }
+            if (props.isTripAdd) {
+                await insertUserTrip(newTrip);
+            } else {
+                await updateUserTrip(newTrip);
+            }
+            props.getUserTrips();
+            setCity('');
+            setCountry('');
+            setEmails([]);
+            setStartDate(new Date());
+            setEndDate(new Date());
+            props.setShowForm(false);
+        }
+    };
+
+    return (
+        <form onSubmit={submitTrip} className={'tripForm'}>
+            <Select options={options} value={country} onChange={changeHandler} className="countrySelect"/>
+            <label for='city'>City</label>
+            <input id='city' type='text' name='city' value={city} placeholder={'Enter city here'} onChange={handleTextChange} className="city_input"/>
```

Comment:
You can use the `AutoComplete` here with `options={{fields:['ALL']}}` (you don't need to specify `types` as the default is `cities`)